lib/ReportDataCollector.php

```diff
@@ -130,20 +139,41 @@ public function getReportJson($options = JSON_PRETTY_PRINT, $depth = 512) {
 	}
 
 	/**
+	 * @param array $report
+	 */
+	public function updateReport(&$report) {
+		foreach ($this->appConfigData as $index) {
+			foreach ($index as $innerKey => $innerVal) {
+				if(in_array($innerKey, array_keys($report))) {
+					$reportIndex = $innerKey;
```

Comment:
I thought the extension point is there to extend the $report array, not manipulate it. If you let it manipulate the existing values, that might cause faulty configreports if a developer tries to mess with it.

In other words: i would like to apply the open/closed principle to this (https://en.wikipedia.org/wiki/Open/closed_principle) :)